### PR TITLE
Unhappy paths for savings pot creation

### PR DESCRIPTION
**Pull request by usamabutt101**
### Why
Deposits - [FE] Savings pot-i: Unhappy paths for savings pot creation 
Please give a short description why these code changes are being made, by replacing this line.
Handle unhappy path flow for create saving pot
Link to Jira ticket: <place here>
https://101digital.atlassian.net/browse/ADBP-1318?atlOrigin=eyJpIjoiOTNhMzI4OTdjYjA5NDcyNDlkZWQzOTBjYzQ0ODYzNjEiLCJwIjoiaiJ9
### ToDo

- [ ]
- [ ]
- [ ]

If this is a `Draft:` or `WIP:` PR, please describe still needed actions, else delete section

### CODE REVIEW

File: src/components/adb-saving-pot-input-name/index.tsx [hunk 1]
```diff
@@ -26,25 +27,30 @@ type ADBSavingPotInputNameProps = {
...
-    keyboardHeight > 0 && Platform.OS === 'ios' ? keyboardHeight : 15;
+    keyboardHeight > 0 && Platform.OS === "ios" ? keyboardHeight : 15;
 
   useEffect(() => {
     const keyboardDidShowListener = Keyboard.addListener(
-      'keyboardDidShow',
+      "keyboardDidShow",
       (e) => {
-        console.log('event', e);
+        console.log("event", e);
```
Comment (sharmal33) [inline src/components/adb-saving-pot-input-name/index.tsx:47]: remove this

File: src/components/adb-saving-pot-input-name/index.tsx [hunk 2]
```diff
@@ -56,33 +62,34 @@ const ADBSavingPotInputNameComponent = (props: ADBSavingPotInputNameProps) => {
...
       <View style={styles.content}>
         <Formik
           innerRef={formikRef}
           enableReinitialize={true}
           initialValues={InputPotNameData.empty()}
-          validationSchema={InputPotNameSchema}
+          validationSchema={validationSchema}
           onSubmit={(values) => {
-            console.log('submit -> values', values);
+            console.log("submit -> values", values);
```
Comment (sharmal33) [inline src/components/adb-saving-pot-input-name/index.tsx:75]: remove this log

File: src/components/adb-saving-pot-input-name/index.tsx [hunk 3]
```diff
@@ -113,10 +124,10 @@ const styles = StyleSheet.create({
     paddingHorizontal: 22,
   },
   columnItem: {
-    justifyContent: 'space-between',
+    justifyContent: "space-between",
   },
   titleCategory: {
-    color: '#1B1B1B',
+    color: "#1B1B1B",
```
Comment (sharmal33) [inline src/components/adb-saving-pot-input-name/index.tsx:130]: please color palette

File: src/components/adb-saving-pot-input-name/index.tsx [hunk 4]
```diff
@@ -127,15 +138,15 @@ const styles = StyleSheet.create({
   categoryItem: {
     width: width / 2.3,
     borderWidth: 1,
-    borderColor: '#000000',
+    borderColor: "#000000",
```
Comment (sharmal33) [inline src/components/adb-saving-pot-input-name/index.tsx:141]: use color palette

File: src/components/adb-saving-pot-input-name/index.tsx [hunk 5]
```diff
@@ -148,35 +159,35 @@ const styles = StyleSheet.create({
   },
   readMoreBtn: {},
   readMoreLabel: {
-    color: '#1B1B1B',
+    color: "#1B1B1B",
```
Comment (sharmal33) [inline src/components/adb-saving-pot-input-name/index.tsx:162]: use color palette

File: src/components/adb-saving-pot-input-name/index.tsx [hunk 6]
```diff
@@ -148,35 +159,35 @@ const styles = StyleSheet.create({
...
     fontFamily: fonts.semiBold,
     fontSize: 14,
     lineHeight: 20,
-    textDecorationLine: 'underline',
+    textDecorationLine: "underline",
     marginLeft: 5,
   },
   noteSection: {
-    backgroundColor: '#F5F5F5',
+    backgroundColor: "#F5F5F5",
```
Comment (sharmal33) [inline src/components/adb-saving-pot-input-name/index.tsx:170]: use color palette

File: src/components/adb-saving-pot-input-name/index.tsx [hunk 7]
```diff
@@ -148,35 +159,35 @@ const styles = StyleSheet.create({
...
     borderRadius: 3,
     padding: 15,
     marginBottom: 12,
-    flexDirection: 'row',
+    flexDirection: "row",
   },
   profitRateContainer: {
     padding: 15,
-    backgroundColor: '#F5F5F5',
+    backgroundColor: "#F5F5F5",
```
Comment (sharmal33) [inline src/components/adb-saving-pot-input-name/index.tsx:178]: use color palette

File: src/components/adb-saving-pot-input-name/index.tsx [hunk 8]
```diff
@@ -148,35 +159,35 @@ const styles = StyleSheet.create({
...
   },
   profitRateTitle: {
-    color: '#858585',
+    color: "#858585",
     fontSize: 12,
     fontFamily: fonts.regular,
   },
   profitRateValue: {
-    color: '#333333',
+    color: "#333333",
```
Comment (sharmal33) [inline src/components/adb-saving-pot-input-name/index.tsx:190]: use color palette

File: src/components/adb-saving-pot-input-name/index.tsx [hunk 9]
```diff
@@ -148,35 +159,35 @@ const styles = StyleSheet.create({
...
-    flexDirection: 'row',
-    justifyContent: 'space-between',
-    alignItems: 'center',
+    flexDirection: "row",
+    justifyContent: "space-between",
+    alignItems: "center",
   },
   profitRateTitle: {
-    color: '#858585',
+    color: "#858585",
```
Comment (sharmal33) [inline src/components/adb-saving-pot-input-name/index.tsx:185]: use color palette

File: src/components/adb-saving-pot-input-name/index.tsx [hunk 10]
```diff
@@ -187,21 +198,21 @@ const styles = StyleSheet.create({
     marginLeft: 5,
   },
   remainingLabel: {
-    textAlign: 'right',
+    textAlign: "right",
   },
   imgPlaceHolderContainer: {
-    backgroundColor: '#D9D9D9',
+    backgroundColor: "#D9D9D9",
```
Comment (sharmal33) [inline src/components/adb-saving-pot-input-name/index.tsx:204]: use color palette

File: src/components/adb-saving-pot-input-name/index.tsx [hunk 11]
```diff
@@ -230,12 +241,12 @@ const styles = StyleSheet.create({
   },
   title: {
     fontSize: 24,
-    color: '#1B1B1B',
+    color: "#1B1B1B",
```
Comment (sharmal33) [inline src/components/adb-saving-pot-input-name/index.tsx:244]: use color palette

File: src/components/adb-saving-pot-input-name/index.tsx [hunk 12]
```diff
@@ -230,12 +241,12 @@ const styles = StyleSheet.create({
...
     fontFamily: fonts.semiBold,
   },
   subTitle: {
     fontSize: 14,
-    color: '#1B1B1B',
+    color: "#1B1B1B",
```
Comment (sharmal33) [inline src/components/adb-saving-pot-input-name/index.tsx:249]: use color palette

File: src/components/adb-saving-pot-input-name/model.ts [hunk 13]
```diff
@@ -1,19 +1,25 @@
...
-    ),
-});
+export const InputPotNameSchema = (wallets: any) => {
+  return Yup.object().shape({
+    potName: Yup.string()
+      .trim()
+      .required("Pot name field cannot be empty.")
+      .max(40, "Pot name can't be longer than 40 characters.")
+      .matches(/^[a-zA-Z0-9 ]+$/, "Please remove invalid special characters.")
+      .test("checkDuplicate", "Pot name has been used.", function (value: any) {
```
Comment (sharmal33) [inline src/components/adb-saving-pot-input-name/model.ts:17]: plz move all strings to translations file

File: src/components/adb-saving-pot-input-review/index.tsx [hunk 14]
```diff
@@ -50,32 +50,44 @@ const ADBSavingPotInputReviewComponent = (
...
+    (w: any) => w.walletId !== selectedWalletId
+  );
 
   useEffect(() => {
     const keyboardDidShowListener = Keyboard.addListener(
-      'keyboardDidShow',
+      "keyboardDidShow",
       (e) => {
-        console.log('event', e);
+        console.log("event", e);
```
Comment (sharmal33) [inline src/components/adb-saving-pot-input-review/index.tsx:84]: remove this log

File: src/components/adb-saving-pot-input-review/index.tsx [hunk 15]
```diff
@@ -88,14 +100,19 @@ const ADBSavingPotInputReviewComponent = (
...
-      console.log('useEffect check -> potName', potName);
-      const newWallet = wallets.find(w => w.bankAccount.goalName && w.bankAccount.goalName === potName);
-      if(newWallet) {
+    if (wallets.length > 0 && formikRef.current && intervalRef.current) {
+      console.log("useEffect check -> wallets", wallets);
+      console.log("useEffect check -> walletsJSON", JSON.stringify(wallets));
+      console.log(
+        "useEffect check -> formikRef.current?.name",
+        formikRef.current.values
+      );
```
Comment (sharmal33) [inline src/components/adb-saving-pot-input-review/index.tsx:109]: remove this

File: src/components/adb-saving-pot-input-review/index.tsx [hunk 16]
```diff
@@ -88,14 +100,19 @@ const ADBSavingPotInputReviewComponent = (
...
+      const { potName } = formikRef.current.values;
+      console.log("useEffect check -> potName", potName);
```
Comment (sharmal33) [inline src/components/adb-saving-pot-input-review/index.tsx:111]: plz remove all logs

File: src/components/adb-saving-pot-input-review/index.tsx [hunk 17]
```diff
@@ -106,30 +123,30 @@ const ADBSavingPotInputReviewComponent = (
           productName: potName,
           productCategory: potCategory,
           productTargetDate: potTargetDate,
-          currencyCode: 'MYR',
+          currencyCode: "MYR",
```
Comment (sharmal33) [inline src/components/adb-saving-pot-input-review/index.tsx:126]: Currency code should be RM. plz contact @khoi101digital

File: src/components/adb-saving-pot-input-review/index.tsx [hunk 18]
```diff
@@ -157,29 +174,30 @@ const ADBSavingPotInputReviewComponent = (
...
-            potName: potName || '',
-            potAmount: potTargetAmount || '',
+            potName: potName || "",
+            potAmount: potTargetAmount || "",
           }}
-          validationSchema={InputPotReviewSchema}
+          validationSchema={validationSchema}
           onSubmit={(values) => {
-            console.log('submit -> values', values);
+            console.log("submit -> values", values);
```
Comment (sharmal33) [inline src/components/adb-saving-pot-input-review/index.tsx:190]: remove logs

File: src/components/adb-saving-pot-input-review/index.tsx [hunk 19]
```diff
@@ -268,9 +286,14 @@ const ADBSavingPotInputReviewComponent = (
                   }}
                 >
                   <ADBButton
-                    label={'Continue'}
+                    disabled={Object.keys(errors).length !== 0}
+                    label={"Continue"}
```
Comment (sharmal33) [inline src/components/adb-saving-pot-input-review/index.tsx:290]: plz move this to translations file

File: src/components/adb-saving-pot-input-review/index.tsx [hunk 20]
```diff
@@ -289,7 +312,7 @@ const ADBSavingPotInputReviewComponent = (
             <View style={styles.calendarContainer}>
               <Calendar
                 theme={{
-                  arrowColor: '#1B1B1B',
+                  arrowColor: "#1B1B1B",
```
Comment (sharmal33) [inline src/components/adb-saving-pot-input-review/index.tsx:315]: use color palette

File: src/components/adb-saving-pot-input-review/index.tsx [hunk 21]
```diff
@@ -313,42 +336,42 @@ const styles = StyleSheet.create({
...
   bsTitle: {
-    color: '#1B1B1B',
+    color: "#1B1B1B",
     fontFamily: fonts.semiBold,
     fontSize: 24,
   },
   calendarContainer: {
     marginVertical: 15,
-    backgroundColor: '#F5F5F5',
+    backgroundColor: "#F5F5F5",
```
Comment (sharmal33) [inline src/components/adb-saving-pot-input-review/index.tsx:349]: use color palette

File: src/components/adb-saving-pot-input-review/index.tsx [hunk 22]
```diff
@@ -361,42 +384,42 @@ const styles = StyleSheet.create({
...
+    justifyContent: "space-between",
+    alignItems: "center",
   },
   noteBold: {
     fontSize: 13,
     lineHeight: 16,
     fontFamily: fonts.bold,
     marginLeft: 5,
-    color: '#1B1B1B',
+    color: "#1B1B1B",
```
Comment (sharmal33) [inline src/components/adb-saving-pot-input-review/index.tsx:414]: use color palette

File: src/components/adb-saving-pot-input-review/index.tsx [hunk 23]
```diff
@@ -433,15 +456,15 @@ const styles = StyleSheet.create({
...
+    flexDirection: "row",
+    alignItems: "center",
+    justifyContent: "space-between",
   },
   buttonAction: {
     padding: 12,
     borderRadius: 10,
     borderWidth: 1,
-    borderColor: '#333',
+    borderColor: "#333",
```
Comment (sharmal33) [inline src/components/adb-saving-pot-input-review/index.tsx:467]: use color palette

File: src/components/adb-saving-pot-input-review/index.tsx [hunk 24]
```diff
@@ -450,12 +473,12 @@ const styles = StyleSheet.create({
   },
   title: {
     fontSize: 24,
-    color: '#1B1B1B',
+    color: "#1B1B1B",
```
Comment (sharmal33) [inline src/components/adb-saving-pot-input-review/index.tsx:476]: use color palette

File: src/components/adb-saving-pot-input-review/model.ts [hunk 25]
```diff
@@ -1,22 +1,26 @@
...
-    .required('Please input your target amount')
-});
+export const InputPotReviewSchema = (wallets: any) => {
+  return Yup.object().shape({
+    potName: Yup.string()
+      .trim()
+      .required("Pot name field cannot be empty.")
+      .max(40, "Pot name can't be longer than 40 characters")
+      .matches(/^[a-zA-Z0-9 ]+$/, "Please remove invalid special characters.")
+      .test("checkDuplicate", "Pot name has been used.", function (value: any) {
```
Comment (sharmal33) [inline src/components/adb-saving-pot-input-review/model.ts:17]: plz move all strings to translations file

File: src/components/adb-saving-pot-input-review/model.ts [hunk 26]
```diff
@@ -1,22 +1,26 @@
...
+        const isDetected = wallets.find(
+          (w: any) =>
+            w.bankAccount?.goalName?.toLowerCase() === value?.toLowerCase()
+        );
+        return isDetected ? false : true;
+      }),
+    potAmount: Yup.string().trim().required("Please input your target amount"),
```
Comment (sharmal33) [inline src/components/adb-saving-pot-input-review/model.ts:24]: plz move all strings to translations file

**Review by sharmal33 — CHANGES_REQUESTED** [review]

File: src/components/adb-saving-pot-input-name/index.tsx (hunk 1 above)
Comment (usamabutt101) [reply src/components/adb-saving-pot-input-name/index.tsx:47]: Sharmal these consoles were not added by me I have enable the prettier thats why my prettier just change the single quotes into double quotes

Comment (usamabutt101) [reply src/components/adb-saving-pot-input-name/index.tsx:47]: removed!

File: src/components/adb-saving-pot-input-name/index.tsx (hunk 2 above)
Comment (usamabutt101) [reply src/components/adb-saving-pot-input-name/index.tsx:75]: removed!

**Pull request merged**